### PR TITLE
Feature/v1 api create and show snapshots

### PR DESCRIPTION
**Pull request by whatsadebugger**
So I believe I got this working almost fully. I am not sure about my test is Assignments Controller. ShowSnapShot.

### CODE REVIEW

File: web/assignments_controller.go [hunk 1]
```diff
@@ -60,3 +60,46 @@ func (ac *AssignmentsController) Show(c *gin.Context) {
...
+
+	if j, err := ac.App.Store.FindJob(id); err == storm.ErrNotFound {
+		c.JSON(404, gin.H{
+			"errors": []string{"Job not found"},
+		})
+	} else if err != nil {
+		c.JSON(500, gin.H{
+			"errors": []string{err.Error()},
+		})
+	} else if jr, err := startJob(j, ac.App.Store, models.JSON{}); err != nil {
```
Comment (se3000) [inline web/assignments_controller.go]: Could you hit this edge case with a unit test? I think the easiest way would be to create a job spec with a endAt which has already passed, and then creating a snapshot for it.

File: web/assignments_controller_test.go [hunk 2]
```diff
@@ -71,3 +72,67 @@ func TestAssignmentsController_Show_V1_Format(t *testing.T) {
...
+
+	cltest.WaitForRuns(t, j, app.Store, 1)
+
+	url = app.Server.URL + "/v1/snapshots/" + runID
+	resp2 := cltest.BasicAuthGet(url)
+	assert.Equal(t, 200, resp2.StatusCode, "Response should be successful")
+
+	var ss models.SnapShot
+	assert.Nil(t, json.Unmarshal(cltest.ParseResponseBody(resp2), &ss))
+
```
Comment (se3000) [inline web/assignments_controller_test.go]: It's possible this will unmarshal without error and still not contain any useful data. Please add an additional check to make sure that this comes out with the keys and values that you expect, probably using something like `assert.JSONEq`.

File: web/snapshots_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	j := cltest.FixtureCreateJobWithAssignmentViaWeb(t, app, "../internal/fixtures/web/v1_format_job_past_endat_time.json")
+
+	url := app.Server.URL + "/v1/assignments/" + j.ID + "/snapshots"
+	resp := cltest.BasicAuthPost(url, "application/json", bytes.NewBuffer([]byte{}))
+
+	assert.Equal(t, 500, resp.StatusCode, "Response should be server error")
+}
+
+func TestSnapshotsController_ShowSnapshot_V1_Format(t *testing.T) {
+	t.Parallel()
```
Comment (whatsadebugger) [inline web/snapshots_controller_test.go]: this test was crazy

Comment (se3000) [reply web/snapshots_controller_test.go]: Haha, yeah. It looks like it's testing a lot of the block height confirmation logic, which is probably out of scope for a unit test of the controller. If you assert that the run reaches the PendingConfirmation state, that'd be a sufficient test, and you could cut out a lot of the ethereum mocking.
